lib/workers/repository/update/branch/execute-post-upgrade-commands.ts

```diff
@@ -1,163 +1,41 @@
-              cwd: GlobalConfig.get('localDir'),
-            });
-
-            logger.debug(
-              { cmd: compiledCmd, ...execResult },
-              'Executed post-upgrade task'
-            );
-          } catch (error) {
-            artifactErrors.push({
-              lockFile: upgrade.packageFile,
-              stderr: sanitize(error.message),
-            });
-          }
-        } else {
-          logger.warn(
-            {
-              cmd,
-              allowedPostUpgradeCommands,
-            },
-            'Post-upgrade task did not match any on allowedPostUpgradeCommands list'
-          );
-          artifactErrors.push({
-            lockFile: upgrade.packageFile,
-            stderr: sanitize(
-              `Post-upgrade command '${cmd}' has not been added to the allowed list in allowedPostUpgradeCommands`
-            ),
-          });
-        }
-      }
-
-      const status = await getRepoStatus();
-
-      for (const relativePath of status.modified.concat(status.not_added)) {
-        for (const pattern of fileFilters) {
-          if (minimatch(relativePath, pattern)) {
-            logger.debug(
-              { file: relativePath, pattern },
-              'Post-upgrade file saved'
-            );
-            const existingContent = await readLocalFile(relativePath);
-            const existingUpdatedArtifacts = updatedArtifacts.find(
-              (ua) => ua.path === relativePath
-            );
-            if (existingUpdatedArtifacts?.type === 'addition') {
-              existingUpdatedArtifacts.contents = existingContent;
-            } else {
-              updatedArtifacts.push({
-                type: 'addition',
-                path: relativePath,
-                contents: existingContent,
-              });
-            }
-            // If the file is deleted by a previous post-update command, remove the deletion from updatedArtifacts
-            updatedArtifacts = updatedArtifacts.filter(
-              (ua) => !(ua.type === 'deletion' && ua.path === relativePath)
-            );
-          }
-        }
-      }
-
-      for (const relativePath of status.deleted || []) {
-        for (const pattern of fileFilters) {
-          if (minimatch(relativePath, pattern)) {
-            logger.debug(
-              { file: relativePath, pattern },
-              'Post-upgrade file removed'
-            );
-            updatedArtifacts.push({
-              type: 'deletion',
-              path: relativePath,
-            });
-            // If the file is created or modified by a previous post-update command, remove the modification from updatedArtifacts
-            updatedArtifacts = updatedArtifacts.filter(
-              (ua) => !(ua.type === 'addition' && ua.path === relativePath)
-            );
-          }
-        }
-      }
-    }
+    updatedArtifacts = result.updatedArtifacts;
+    artifactErrors = [...artifactErrors, ...result.artifactErrors];
```

Comment:
we should pass the array to the called function and add results directly, so we don't need to clone it in each iteration.